capstone-backend/src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -216,13 +218,15 @@ static VolumeData individualBookToVolumeData(JsonElement singleBook) {
     JsonElement thumbnailElement = volumeInfoObj.get("imageLinks");
     ArrayList<String> sizes = new ArrayList<>(Arrays.asList("extraLarge", "large", "medium", "small", "thumbnail"));
 
-    String thumbnailLink = "";
-    Optional<String> greatestSize = sizes.stream().filter(size -> thumbnailElement.getAsJsonObject().get(size) != null)
-        .findFirst();
-    if (greatestSize.isPresent()) {
-      thumbnailLink = thumbnailElement.getAsJsonObject().get(greatestSize.get().toString()).getAsString();
+    String thumbnailLink = DEFAULT_THUMBNAIL_IMAGE;
+    if (thumbnailElement != null) {
+      Optional<String> greatestSize = sizes.stream()
+          .filter(size -> thumbnailElement.getAsJsonObject().get(size) != null).findFirst();
+      if (greatestSize.isPresent()) {
+        thumbnailLink = thumbnailElement.getAsJsonObject().get(greatestSize.get().toString()).getAsString();
+      }
+      thumbnailLink = thumbnailLink.replace("http", "https");
```

Comment:
If this somehow executes with greatestSize not being present, there may be an issue replacing the `http` within the DEFAULT_THUMBNAIL_IMAGE where it may end up producing something like `httpss`

Reply:
Could you instead have `DEFAULT_THUMBNAIL_IMAGE` use `http` to begin with, and move the replace down below the `if` statement? This way regardless of the presence of `greatestSize`, it will always replace `http` with `https`?

Reply:
Could you avoid this by replacing "http:" with "https:"?

Are all the images available over https as well or does this sometimes result in the default image?

Reply:
Yes Geoff that change would make it so I do not have to worry about this check. I have now changed it to do `http:` to `https:`. I'm not sure why I did not do it this way to start.

I have not run into the case where books are offered only over http and not https. Even when I paste these book links into my browser directly from the API, Google routes me to the https site. This https replace is to avoid the frontend error of serving images from an insecure source.